src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -715,6 +715,7 @@ internal static bool IsFieldOrPropertyInitializer(BoundStatement initializer)
                 {
                     case SyntaxKind.VariableDeclarator:
                     case SyntaxKind.PropertyDeclaration:
```

Comment:
>SyntaxKind.PropertyDeclaration [](start = 25, length = 30)

There is a number of references to PropertyDeclaration in instrumenters (InstrumentFieldOrPropertyInitializer, etc.) that I think should be adjusted as well. I think we should audit all checks like that (both type and kind) and make necessary adjustments.